tests/components/roon/test_config_flow.py

```diff
@@ -3,164 +3,217 @@
-            result["flow_id"], {"host": "1.1.1.1"}
-        )
-        result2 = await hass.config_entries.flow.async_configure(
-            result["flow_id"], user_input={}
+    ):
+
+        await setup.async_setup_component(hass, "persistent_notification", {})
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
         )
         await hass.async_block_till_done()
 
-    assert result2["type"] == "create_entry"
-    assert result2["title"] == "Roon Labs Music Player"
-    assert result2["data"] == {"host": "1.1.1.1", "api_key": "good_token"}
-    assert len(mock_setup.mock_calls) == 1
-    assert len(mock_setup_entry.mock_calls) == 1
-
-
-async def test_form_no_token(hass):
-    """Test we handle no token being returned (timeout or not authorized)."""
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": config_entries.SOURCE_USER}
-    )
-    with patch("homeassistant.components.roon.config_flow.TIMEOUT", 0,), patch(
-        "homeassistant.components.roon.const.AUTHENTICATE_TIMEOUT",
-        0,
-    ), patch(
-        "homeassistant.components.roon.config_flow.RoonApi",
-        return_value=RoonApiMock(None),
-    ):
-        await hass.config_entries.flow.async_configure(
-            result["flow_id"], {"host": "1.1.1.1"}
-        )
+        # Should go straight to link if server was discovered
+        assert result["type"] == "form"
+        assert result["step_id"] == "link"
+        assert result["errors"] == {}
 
         result2 = await hass.config_entries.flow.async_configure(
             result["flow_id"], user_input={}
         )
+        await hass.async_block_till_done()
 
-    assert result2["type"] == "form"
-    assert result2["errors"] == {"base": "invalid_auth"}
+    assert result2["title"] == "Roon Labs Music Player"
+    assert result2["data"] == {
+        "host": None,
+        "api_key": "good_token",
+        "roon_server_id": "core_id",
+    }
 
 
-async def test_form_unknown_exception(hass):
-    """Test we handle cannot connect error."""
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": config_entries.SOURCE_USER}
-    )
+async def test_unsuccessful_discovery_user_form_and_auth(hass):
+    """Test unsuccessful discover, user adding the host via the form and then successful auth."""
 
     with patch(
         "homeassistant.components.roon.config_flow.RoonApi",
-        side_effect=Exception,
+        return_value=RoonApiMock(),
+    ), patch(
+        "homeassistant.components.roon.config_flow.RoonDiscovery",
+        return_value=RoonDiscoveryFailedMock(),
+    ), patch(
+        "homeassistant.components.roon.async_setup", return_value=True
+    ), patch(
+        "homeassistant.components.roon.async_setup_entry",
+        return_value=True,
     ):
+
+        await setup.async_setup_component(hass, "persistent_notification", {})
```

Comment:
The persistent notification integration doesn't need to be set up inside the patch context manager.

Reply:
Side note: We usually set up the persistent notification integration in the beginning of the config flow tests to avoid a log error about missing service. It won't affect the tests of the config flow, it just cleans up the log, making it easier to troubleshoot.

Reply:
Understood - shall I drop this into the PR? (just aware you've already approved).

Reply:
Yeah, please do that. Then we'll trigger another build to hopefully get it passed the unrelated flaky test.